docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,214 @@
+```
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG FLAG			VALUE		DESCRIPTION		MENU PATH
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG_ARMV7M_MPU		y		MPU support		Chip Selection -> MPU Support
+	CONFIG_ARMV7M_MPU_NREGIONS	16		Number of MPU regions	Chip Selection -> MPU Support -> Number of MPU Regions
+	------------------------------------------------------------------------------------------------------------------------------
+```
+
+- Similarly, for ARMv8M architecture, below MPU configs should be enabled.
+```
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG FLAG			VALUE		DESCRIPTION		MENU PATH
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG_ARMV8M_MPU		y		MPU support		Chip Selection -> MPU Support
+	CONFIG_ARMV8M_MPU_NREGIONS	8		Number of MPU regions	Chip Selection -> MPU Support -> Number of MPU Regions
+	------------------------------------------------------------------------------------------------------------------------------
+```
+
+## How to control MPU
+1. **How to add new MPU region**
+	- Logical mapping of MPU regions to their respective memory segments is done statically using below enumeration in common mpu.h header file (os/include/tinyara/mpu.h).
+	```c
+	enum MPU_REG_NUM {
+	#ifdef CONFIG_APP_BINARY_SEPARATION
+	#ifdef CONFIG_SUPPORT_COMMON_BINARY
+	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+		MPU_REG_NUM_COM_LIB_TXT,	/* Common Binary */
+		MPU_REG_NUM_COM_LIB_RO,
+		MPU_REG_NUM_COM_LIB_DATA,
+	#else
+		MPU_REG_NUM_COM_LIB,
+	#endif
+	#endif
+	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+		MPU_REG_NUM_APP_TXT,		/* Apps */
+		MPU_REG_NUM_APP_RO,
+		MPU_REG_NUM_APP_DATA,
+	#else
+		MPU_REG_NUM_APP,
+	#endif
+	#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+		MPU_REG_NUM_STK,		/* Stack */
+	#endif
+	#else
+		MPU_REG_NUM_UFLASH,
+		MPU_REG_NUM_URAM,
+	#endif
+		MPU_REG_NUM_MAX
+	};
+	```
+	- The ordering of MPU regions in this enum is kept as above taking into consideration below rule:
+		- For overlapping address range between MPU regions, attributes of higher MPU region number will override those set by lower numbered MPU region.
+	- Your new MPU region should be added the enum above.
+2. **How to register MPU region to real MPU hardware**
+	- Below list of functions is available in TizenRT for registering MPU region to MPU hardware.
+		- Several functions of mpu_xxxx format are available in architecture specific mpu.h file (eg. os/arch/arm/src/armv7-m/mpu.h) for one-time configuration of MPU regions.
+			- Example: For configuring MPU regions for User Flash and User Internal SRAM, below functions are available.
+			```c
```

Comment:
I found the doc which describes the MPU functions of ARMv7-R, https://github.com/Samsung/TizenRT/blob/master/docs/HowToUseMPU.md.
Could you give a link here?
```
[Description of MPU APIs on ARMv7-R](https://github.com/Samsung/TizenRT/blob/master/docs/HowToUseMPU.md)
```

Reply:
If you think that providing the description of only armv7-r is strange, we could delete that. And we could give the file links for 7m, 7r and 8m.

Reply:
I have provided mpu.h file links (that describe MPU functions) for armv7-m, armv7-r and armv8-m in the doc.